catalog/transformer/catalog-transformer-preview/src/main/java/org/codice/ddf/transformer/preview/PreviewMetacardTransformer.java

```diff
@@ -34,15 +56,80 @@ public BinaryContent transform(Metacard metacard, Map<String, Serializable> argu
     }
 
     String preview = "No preview text available.";
+
     if (metacard.getAttribute(Extracted.EXTRACTED_TEXT) != null
         && metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue() != null) {
       preview =
           StringEscapeUtils.escapeHtml4(
                   metacard.getAttribute(Extracted.EXTRACTED_TEXT).getValue().toString())
               .replaceAll("[\n|\r]", "<br>");
-      preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
+    } else if (previewFromMetadata
+        && xPathExpressions != null
+        && StringUtils.isNotEmpty(metacard.getMetadata())) {
+      String text =
+          xPathExpressions
+              .stream()
+              .map(x -> evaluateXPath(metacard.getMetadata(), x))
+              .filter(Objects::nonNull)
+              .findFirst()
+              .orElse("");
+      if (StringUtils.isNotEmpty(text)) {
+        preview = StringEscapeUtils.escapeHtml4(text).replaceAll("[\n|\r]", "<br>");
+      }
     }
 
+    preview = String.format("<head><meta charset=\"utf-8\"/>%s</head>", preview);
+
     return new BinaryContentImpl(IOUtils.toInputStream(preview));
   }
+
+  private String evaluateXPath(String metadata, XPathExpression xPathExpression) {
+    StringReader metadataReader = new StringReader(metadata);
+    InputSource inputXml = new InputSource(metadataReader);
+
+    String text = null;
+    try {
+      Node result = (Node) xPathExpression.evaluate(inputXml, XPathConstants.NODE);
+      if (result != null && result.getTextContent() != null) {
+        text = result.getTextContent().trim();
+      }
+    } catch (XPathExpressionException e) {
+      LOGGER.error("Error evaluating xpath");
+    }
+
+    return text;
+  }
+
+  private List<XPathExpression> createXPathExpressions(List<String> previewElements) {
+    XPathFactory xPathFactory = XPathFactory.newInstance();
+    List<XPathExpression> xPathExpressions = new ArrayList<>();
+
+    String xPathString = "//*[name()='%s']";
+
+    try {
+      for (String element : previewElements) {
+        xPathExpressions.add(xPathFactory.newXPath().compile(String.format(xPathString, element)));
+      }
+    } catch (XPathExpressionException e) {
```

Comment:
If an xpath expression fails to compile, the method will not try to compile any remaining other elements in the `previewElements` collection.I don't think that is what you want (but I'm not sure). 

Create a new method to call from inside the for loop and handle the exception there:

```
XPathExpressionException matchAllElementsIgnoreNamespace(String elementName) {
   try {
       String xexpr = String.format("//*[name()='%s']", elementName);
        return  XPathFactory.newInstance().newPath().compile(xexpr);    
    }
 } catch (XPathExpressionException e) {
    LOGGER.debug("Bad XPath expression. Expected an element named {}", elementName);
   //Return null or Optional and check for it in the method that calls this one.
}
```